ui/src/app/settings/reducers/config.js

```diff
@@ -5,24 +5,27 @@ const config = produce(
     switch (action.type) {
       case "FETCH_CONFIG_START":
         draft.loading = true;
-        return;
+        break;
       case "FETCH_CONFIG_SUCCESS":
         draft.loading = false;
         draft.loaded = true;
         draft.items = action.payload;
-        return;
+        break;
       case "UPDATE_CONFIG_START":
         draft.saving = true;
-        return;
+        break;
       case "UPDATE_CONFIG_SUCCESS":
         draft.saving = false;
+        break;
+      case "UPDATE_CONFIG_SYNC":
+        draft.synced = true;
```

Comment:
Actually, not sure about setting synced here and we'd have to manually reset it. This is not in response to anything and it might be better to only do this on a `_SUCCESS` message.

Doing it here will also mean that you could get false positives if you're waiting for a sync and a different update comes over the socket.

Reply:
In practical terms the UI can probably respond to a SUCCESS, but that makes the UI eventually consistent which is maybe a bit weird.